### PR TITLE
Upgrade to sbt 1.1.1

### PR DESCRIPTION
**Pull request by by-dam**

### CODE REVIEW

**Comment by carlpulley** [comment]
Is it sensible to think about doing this yet?

My issue is that SBT 1.X is very new and tons of issues have been uncovered with this (some have been fixed, but other serious issues will lurk and cause lost time on projects debugging them). So I'd argue that we should wait longer before this ticket is merged.

If there is a pressing need for this work, then this code should be cross-SBT compiled so that other teams may use it.

File: src/main/scala/net/cakesolutions/CakeBuildInfoPlugin.scala [hunk 1]
```diff
@@ -30,13 +31,10 @@ object CakeBuildInfoPlugin extends AutoPlugin {
   val autoImport = CakeBuildInfoKeys
   import autoImport._
 
-  /** @see http://www.scala-sbt.org/0.13/api/index.html#sbt.package */
```
Comment (carlpulley) [inline src/main/scala/net/cakesolutions/CakeBuildInfoPlugin.scala]: Why are we removing the Scaladoc?

File: src/main/scala/net/cakesolutions/CakeBuildPlugin.scala [hunk 2]
```diff
@@ -184,12 +180,12 @@ object CakeBuildKeys {
         testForkedParallel in IntegrationTest := false,
         testGrouping := {
           val opts = ForkOptions(
-            bootJars = Nil,
             javaHome = javaHome.value,
-            connectInput = connectInput.value,
             outputStrategy = outputStrategy.value,
-            runJVMOptions = javaOptions.value,
-            workingDirectory = Some(baseDirectory.value),
+            bootJars = Nil.toVector,
```
Comment (carlpulley) [inline src/main/scala/net/cakesolutions/CakeBuildPlugin.scala]: Surely `Vector()` or better `Vector.empty`?

File: src/main/scala/net/cakesolutions/CakePublishMavenPlugin.scala [hunk 3]
```diff
@@ -61,20 +59,21 @@ object CakePublishMavenPlugin extends AutoPlugin {
 
     def addPublishingCredentials(): Setting[Task[Seq[Credentials]]] = {
       credentials ++= {
-        if ((Path.userHome / ".sbt" / "0.13" / ".credentials").exists()) {
-          Seq(Credentials(Path.userHome / ".sbt" / "0.13" / ".credentials"))
+        if ((Path.userHome / ".sbt" / "1.0" / ".credentials").exists()) {
```
Comment (carlpulley) [inline src/main/scala/net/cakesolutions/CakePublishMavenPlugin.scala]: Can't we factor these SBT version numbers away from here? Changing the SBT version value shouldn't have a surprising change here.

**Comment by simonsouter** [comment]
Might be worth keeping this on a separate 1.x branch of sbt-cake and releasing it separately?  Until we decide to migrate over en masse.

File: src/sbt-test/sbt-cake/testrunner/build.sbt [hunk 4]
```diff
@@ -18,9 +20,7 @@ val testrunner = (project in file("."))
     CakeDockerPlugin,
     CakeTestRunnerPlugin)
   .settings(
-    dependencyOverrides ++= Set(
-      "com.typesafe.akka" %% "akka-actor" % "2.4.20"
-    )
+    dependencyOverrides += "com.typesafe.akka" %% "akka-actor" % "2.4.20"
```
Comment (carlpulley) [inline src/sbt-test/sbt-cake/testrunner/build.sbt]: Shouldn't this value be updated?

File: src/main/scala/net/cakesolutions/CakePublishMavenPlugin.scala (hunk 3 above)
Comment (by-dam) [reply src/main/scala/net/cakesolutions/CakePublishMavenPlugin.scala]: Do you mean getting the credentials directly in `~/.sbt` ?
`(Path.userHome / ".sbt" / ".credentials")`

File: src/sbt-test/sbt-cake/testrunner/build.sbt (hunk 4 above)
Comment (by-dam) [reply src/sbt-test/sbt-cake/testrunner/build.sbt]: Akka version didn't change in this PR so I guess not

File: project/project/CakePlatformDependencies.scala [hunk 5]
```diff
@@ -88,7 +87,7 @@ object CakePlatformDependencies {
   }
 
   object Netty4 {
-    val version = "4.0.45.Final"
+    val version = "4.1.20.Final"
```
Comment (emilypi) [inline project/project/CakePlatformDependencies.scala]: `4.1.21.Final` is out, and stable.

File: project/project/CakePlatformDependencies.scala [hunk 6]
```diff
@@ -99,21 +98,21 @@ object CakePlatformDependencies {
...
+      "com.typesafe.sbt" % "sbt-native-packager" % "1.3.2"
+    val pgp: ModuleID = "com.jsuereth" % "sbt-pgp" % "1.1.0"
     val scalafmt: ModuleID = "com.lucidchart" % "sbt-scalafmt-coursier" % "1.15"
     val scalastyle: ModuleID =
       "org.scalastyle" %% "scalastyle-sbt-plugin" % "1.0.0"
     val scoverage: ModuleID = "org.scoverage" % "sbt-scoverage" % "1.5.1"
-    val sonatype: ModuleID = "org.xerial.sbt" % "sbt-sonatype" % "2.0"
-    val wartRemover: ModuleID = "org.wartremover" % "sbt-wartremover" % "2.1.1"
+    val sonatype: ModuleID = "org.xerial.sbt" % "sbt-sonatype" % "2.1"
+    val wartRemover: ModuleID = "org.wartremover" %% "sbt-wartremover" % "2.2.1"
```
Comment (emilypi) [inline project/project/CakePlatformDependencies.scala:115]: We should plan on incorporating `scalafix` as well

File: src/sbt-test/sbt-cake/release-notes/build.sbt [hunk 7]
```diff
@@ -17,7 +19,7 @@ snapshotRepositoryResolver :=
 repositoryResolver :=
   Some(Resolver.file("file", new File(s"${target.value}/release")))
 
-issueManagementUrl := Some(new URL("http://localhost:8080"))
+issueManagementUrl := Some(new URL(s"http://localhost:8080"))
```
Comment (emilypi) [inline src/sbt-test/sbt-cake/release-notes/build.sbt]: why `s`?

**Comment by emilypi** [comment]
I can make another pass at this over the weekend. There are a few updates I'd like to request.

File: project/project/CakePlatformDependencies.scala (hunk 6 above)
Comment (by-dam) [reply project/project/CakePlatformDependencies.scala:115]: I agree but would do in another PR

**Review by emilypi — APPROVED** [review]

**Comment by emilypi** [comment]
@fommil if we get another set of approvals, I think we're good to merge this one.

**Comment by fommil** [comment]
How many approvals does it need? I have merge rights and probably publish rights, but that might be automated. Say the word.

**Comment by emilypi** [comment]
If you're good with it, we can merge. Usually it's 2 on the most recent changes, not 2 in general, but I think for this, we're good to merge. The sooner the better.

**Comment by fommil** [comment]
I'll give @carlpulley a day to object. If there is silence, I'll merge. If somebody else merges, then my hands are clean :stuck_out_tongue:

**Comment by emilypi** [comment]
@carlpulley @fommil @simonsouter @by-dam Can we merge this, or are we waiting on something in particular?

**Comment by by-dam** [comment]
I'm ok to merge but @carlpulley had concerns about sbt 1.x maturity.

**Pull request merged**